src/editor.rs

```diff
@@ -291,7 +289,8 @@ impl Editor {
     fn update_window_size(&mut self) -> Result<(), Error> {
         let wsize = sys::get_window_size().or_else(|_| terminal::get_window_size_using_cursor())?;
         // Make room for the status bar and status message
-        (self.screen_rows, self.window_width) = (wsize.0.saturating_sub(2), wsize.1);
+        (self.screen_rows, self.window_width) =
+            (wsize.0.saturating_sub(1 + (self.status_msg().len() + wsize.1) / wsize.1), wsize.1);
```

Comment:
Should it be

```suggestion
            (wsize.0.saturating_sub(1 + (self.status_msg().len() + wsize.1 - 1) / wsize.1), wsize.1);
```

instead? Currently, `1 + (self.status_msg().len() + wsize.1) / wsize.1` is equivalent to `2 + self.status_msg().len() / wsize.1`

Reply:
I think you need at least one extra line

```diff
-        (self.screen_rows, self.window_width) =
-            (wsize.0.saturating_sub(1 + (self.status_msg().len() + wsize.1 - 1) / wsize.1), wsize.1);
+        self.screen_rows =
+            wsize.0.saturating_sub(1 + (self.status_msg().len() + wsize.1 - 1) / wsize.1);
+        self.window_width = wsize.1;
```

At least until [`div_ceil` ](https://doc.rust-lang.org/std/primitive.i32.html#method.div_ceil) is stabilized :)

I just merged #229, which cuts down the total line count, so that should be fine now